assets/src/components/cd/clusters/ClusterUpgradeFlyover.tsx

```diff
@@ -0,0 +1,426 @@
+import {
+  Accordion,
+  AppIcon,
+  Button,
+  ClusterIcon,
+  Flyover,
+  ListBoxItem,
+  MarketPlusIcon,
+  Select,
+  SuccessIcon,
+  Table,
+  Tooltip,
+  WrapWithIf,
+} from '@pluralsh/design-system'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import styled, { useTheme } from 'styled-components'
+import isEmpty from 'lodash/isEmpty'
+import {
+  ClustersRowFragment,
+  RuntimeServicesQuery,
+  useCreatePullRequestMutation,
+  useRuntimeServicesQuery,
+  useUpdateClusterMutation,
+} from 'generated/graphql'
+import {
+  isUpgrading,
+  nextSupportedVersion,
+  supportedUpgrades,
+  toNiceVersion,
+  toProviderSupportedVersion,
+} from 'utils/semver'
+import { ColWithIcon } from 'components/utils/table/ColWithIcon'
+import { Confirm } from 'components/utils/Confirm'
+import { ApolloError } from '@apollo/client'
+import { coerce } from 'semver'
+
+import { createColumnHelper } from '@tanstack/react-table'
+
+import { IconProps } from '@pluralsh/design-system/dist/components/icons/createIcon'
+
+import { GqlError } from '../../utils/Alert'
+import { TabularNumbers } from '../../cluster/TableElements'
+
+import RuntimeServices, {
+  getClusterKubeVersion,
+} from './runtime/RuntimeServices'
+import { deprecationsColumns } from './deprecationsColumns'
+
+const supportedVersions = (cluster: ClustersRowFragment | null) =>
+  cluster?.provider?.supportedVersions?.map((vsn) => coerce(vsn)?.raw) ?? []
+
+const columnHelperUpgrade = createColumnHelper<ClustersRowFragment>()
+
+function ClusterUpgradePr({ prs, setError }) {
+  const pr = prs[0]
+  const [mutation, { loading, error }] = useCreatePullRequestMutation({
+    variables: {
+      id: pr.id,
+      branch: 'mjg/upgrade',
+      context: JSON.stringify({ version: '1.28' }),
+    },
+    onCompleted: (data) => {
+      const url = data.createPullRequest?.url
+
+      if (url) {
+        window.open(url, '_blank')?.focus()
+      }
+    },
+  })
+
+  useEffect(() => {
+    setError(error)
+  }, [error, setError])
+
+  return (
+    <Button
+      type="button"
+      secondary
+      onClick={mutation}
+      loading={loading}
+    >
+      Upgrade Now
+    </Button>
+  )
+}
+
+function ClustersUpgradeNow({
+  cluster,
+  targetVersion,
+  apiDeprecations,
+  refetch,
+  setError,
+}: {
+  cluster?: ClustersRowFragment | null
+  targetVersion: Nullable<string>
+  apiDeprecations: unknown[]
```

Comment:
why is this typed as `unknown[]`?

Reply:
ah good catch- was a placeholder for the linter. fixed